typescript/eks/cluster/index.ts

```diff
@@ -0,0 +1,44 @@
+import autoscaling = require('@aws-cdk/aws-autoscaling');
+import iam = require('@aws-cdk/aws-iam');
+import ec2 = require('@aws-cdk/aws-ec2');
+import eks = require('@aws-cdk/aws-eks');
+import cdk = require('@aws-cdk/core');
+
+class EKSCluster extends cdk.Stack {
+  constructor(scope: cdk.App, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    const vpc = new ec2.Vpc(this, 'EKSVpc');  // Create a new VPC for our cluster
+    
+    // IAM role for our EC2 worker nodes
+    const workerRole = new iam.Role(this, 'EKSWorkerRole', {
+      assumedBy: new iam.ServicePrincipal('ec2.amazonaws.com')
+    });
+
+    const eksCluster = new eks.Cluster(this, 'Cluster', {
+      clusterName: id,
```

Comment:
It's an anti-pattern to specify a cluster name because any replacement-requiring updates will fail on it.

```suggestion
```